src/System.Device.Gpio/System/Device/Gpio/GpioController.Linux.cs

```diff
@@ -34,7 +35,22 @@ private static GpioDriver GetBestDriverForBoard()
             Regex regex = new Regex(@"Hardware\s*:\s*(.*)");
             foreach (string cpuInfoLine in cpuInfoLines)
             {
-                Match match = regex.Match(cpuInfoLine);
+                Match match;
+
+                // a first chance of finding a raspberry pi..... perhaps not on Raspbian. note
+                // that we want this to fail quietly if something goes wrong with the test and continue
+                // with detecing a PI usiing /dev/cpuinfo
+                try
+                {
+                    if (File.ReadAllText(ModelInfoPath).StartsWith("Raspberry Pi"))
+                    {
+                        return new RaspberryPi3Driver();
+                    }
+                }
+                catch { }
```

Comment:
shouldn't this be done before the cpuInfoLines are even read and then spun through? Assuming you mean this to be an optimisation so do this first before any other pattern matching

Reply:
> shouldn't this be done before the cpuInfoLines are even read and then spun through? Assuming you mean this to be an optimisation so do this first before any other pattern matching

Thanks @mjjames. It must a Friday thing.

It even passed testing.

Correction coming up.